snap/quota/quota.go

```diff
@@ -271,11 +272,44 @@ func (grp *Group) getAllowedCPUs() []int {
 	return grp.CPULimit.AllowedCPUs
 }
 
+// GetCPUSetQuota returns the currently active CPU set quota for this group, which
+// could originate from a parent
+func (grp *Group) GetCPUSetQuota() []int {
+	if grp.CPULimit != nil && len(grp.CPULimit.AllowedCPUs) != 0 {
+		return grp.CPULimit.AllowedCPUs
+	}
+
+	parent := grp.parentGroup
+	for parent != nil {
+		if parent.CPULimit != nil && len(parent.CPULimit.AllowedCPUs) != 0 {
+			return parent.CPULimit.AllowedCPUs
+		}
+		parent = parent.parentGroup
+	}
+	return []int{}
+}
+
+// GetCorrectedCPUCount returns the maximum number of allowed CPU cores for
+// this group. It needs to take into account that CPU set might have been set
+// to limit the number of cores, or a direct limit on the number of cores.
+// Goal is to select the most restrictive limit.
+func (grp *Group) GetCorrectedCPUCount() int {
+	cpuCount := runtime.NumCPU()
```

Comment:
Don't we need to mock it for the tests (i.e. `var runtimeNumCPU = runtime.NumCPU` + a mocking helper), otherwise it's going to fail existing tests on boards with low cpu count? We also need a test that mocks such low cpu count.

Reply:
Probably a great idea, but why would this not work on a cpu with 1 core? Then the function will return cpuCount=1 which is correct?

Reply:
You are absolutely correct!

Reply:
This is added now :+1: